crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -146,6 +147,56 @@ impl Format<PyFormatContext<'_>> for FormatDanglingComments<'_> {
     }
 }
 
+/// Formats the content of the passed comment.
+///
+/// * Adds a whitespace between `#` and the comment text except if the first character is a `#`, `:`, `'`, or `!`
+/// * Replaces non breaking whitespaces with regular whitespaces except if in front of a `types:` comment
+const fn format_comment(comment: &SourceComment) -> FormatComment {
+    FormatComment { comment }
+}
+
+struct FormatComment<'a> {
+    comment: &'a SourceComment,
+}
+
+impl Format<PyFormatContext<'_>> for FormatComment<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let slice = self.comment.slice();
+        let comment_text = slice.text(SourceCode::new(f.context().contents()));
+
+        let (content, mut start_offset) = comment_text
+            .strip_prefix('#')
+            .map_or((comment_text, TextSize::new(0)), |rest| {
```

Comment:
When would our comment not start with a `#`

Reply:
This should never be the case, but I prefer to handle it gracefully rather than panicking.

Reply:
i think we should surface that as a log error

Reply:
In my view this is fine because the formatter knows how to recover from it. 

The alternative is to return an `Err(FormatError::SyntaxError)` and abort the formatting.

Reply:
yeah, throwing an error would imho be best for cases that should never happen